File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.kt
class Solution {
    fun largestOddNumber(s: String): String {
        return (s.length-1 downTo 0).
        find { (s[it] - '0') and 1 == 1 }
        ?.let { s.substring(0,it+1) } ?: ""
    }
}

File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.cpp
class Solution {
public:
    string largestOddNumber(string s) {
        int n = s.size();
        for (int i=n-1; i>=0; i--) {
            if ((s[i]-'0')&1) return s.substr(0,i+1);
        }
        return "";
    }
};